plutus-core/plutus-core/src/PlutusCore/Builtin/Runtime.hs

```diff
@@ -131,6 +161,10 @@ newtype BuiltinsRuntime fun val = BuiltinsRuntime
 
 deriving newtype instance (NFData fun) => NFData (BuiltinsRuntime fun val)
 
+instance NoThunks (BuiltinsRuntime fun val) where
+    wNoThunks ctx (BuiltinsRuntime arr) = allNoThunks (noThunks ctx <$> elems arr)
```

Comment:
Maybe send a PR for `instance NoThunks (Array i e)` upstream? It's a `base` type, reasonable to add an instance to the library IMO.